presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -424,6 +404,60 @@ protected Node visitShowPartitions(ShowPartitions showPartitions, Void context)
                     showPartitions.getLimit());
         }
 
+        private List<String> getPartitioningColumns(
+                ShowPartitions showPartitions,
+                QualifiedObjectName table,
+                TableHandle tableHandle,
+                Optional<TableLayout> layoutFromQuery,
+                Map<ColumnHandle, String> columnHandleNames)
+        {
+            TableLayout tableLayout;
+            if (layoutFromQuery.isPresent()) {
+                tableLayout = layoutFromQuery.get();
+            }
+            else {
+                List<TableLayoutResult> layouts = metadata.getLayouts(session, tableHandle, Constraint.alwaysTrue(), Optional.empty());
```

Comment:
I need your help in understanding. It seems, in case of hive connector, this is going:
- call  `com.facebook.presto.hive.HiveMetadata#getTableLayouts`
- call `com.facebook.presto.hive.HivePartitionManager#getPartitions`
- fail with `Query over table '%s' can potentially read more than %s partitions`.

 Isn't it the case here?

Reply:
`com.facebook.presto.hive.TestHiveIntegrationSmokeTest#testPartitionPerScanLimit()` should cover that.